neuraxle/data_container.py

```diff
@@ -497,6 +547,61 @@ def concat(self, data_container: DataContainer):
         return self
 
 
+def _pad_or_keep_incomplete_batch(
+        data_container,
+        batch_size,
+        default_value_data_inputs,
+        default_value_expected_outputs
+) -> 'DataContainer':
+    keep_incomplete_batch = isinstance(default_value_data_inputs, AbsentValuesNullObject)
+
+    if not keep_incomplete_batch:
+        data_container = _pad_incomplete_batch(
+            data_container,
+            batch_size,
+            default_value_data_inputs,
+            default_value_expected_outputs
+        )
+
+    return data_container
+
+
+def _pad_incomplete_batch(
+        data_container: 'DataContainer',
+        batch_size: int,
+        default_value_data_inputs: Any,
+        default_value_expected_outputs: Any
+) -> 'DataContainer':
+    data_container = DataContainer(
+        summary_id=data_container.summary_id,
+        current_ids=_pad_data(
+            data_container.current_ids,
+            default_value=None,
+            batch_size=batch_size
+        ),
+        data_inputs=_pad_data(
+            data_container.data_inputs,
+            default_value=default_value_data_inputs,
+            batch_size=batch_size
+        ),
+        expected_outputs=_pad_data(
+            data_container.expected_outputs,
+            default_value=default_value_expected_outputs,
+            batch_size=batch_size
+        )
+    )
+
+    return data_container
+
+
+def _pad_data(data: Union[List, np.ndarray], default_value: Any, batch_size: int):
```

Comment:
Please let's never assume the precise type of data we're dealing with to make the library open to wide data types, such as a custom object that could represent a sequence of transactions, for instance... 

```suggestion
def _pad_data(data: Iterable, default_value: Any, batch_size: int):
```